File: ios/Classes/entities/SadadTransactionResult.swift
import Foundation

enum SadadTransactionResultStatus {
    case Success, ApiField, Canceled, BackPressed, TransactionFailed;
}

class SadadTransactionResult{
    let status: SadadTransactionResultStatus;
    let message: String;
    let transactionNumber: String?
    
    init(_ status: SadadTransactionResultStatus,_ message: String,_ transactionNumber: String?){
        self.status = status;
        self.message = message;
        self.transactionNumber = transactionNumber;
    }
}

File: android/src/main/kotlin/com/example/sadad_flutter_sdk/entities/SadadTransactionResult.kt
package com.example.sadad_flutter_sdk.entities

enum class SadadTransactionResultStatus {
    Success, ApiField, Canceled, BackPressed, TransactionFailed;
}

data class SadadTransactionResult(
    val status: SadadTransactionResultStatus,
    val message: String,
    val transactionNumber: String?
);